Add codebase doc (ECOOP23 artifact version) (#7)
LPTK/mlscript: doc/mls-codebase-doc.md

> # Document to the Codebase of the MLscript Compiler

Reviewer: It's *documentation*, not *document*... Please also correct below.

> This is the document to the codebase of the MLscript Compiler for the reuse
> of our MLscript implementation.

Reviewer: ```suggestion
This is the documentation of the MLscript codebase.
```
